nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/GetSNMP.java

```diff
@@ -16,184 +16,188 @@
  * {@link #onTrigger(ProcessContext, ProcessSession)} method, will construct a
  * {@link FlowFile} containing in its properties the information retrieved.
  * The output {@link FlowFile} won't have any content.
  */
-@Tags({ "snmp", "get", "oid", "walk" })
+@Tags({"snmp", "get", "oid", "walk"})
 @InputRequirement(Requirement.INPUT_FORBIDDEN)
 @CapabilityDescription("Retrieves information from SNMP Agent and outputs a FlowFile with information in attributes and without any content")
 @WritesAttributes({
-    @WritesAttribute(attribute=SNMPUtils.SNMP_PROP_PREFIX + "*", description="Attributes retrieved from the SNMP response. It may include:"
-            + " snmp$errorIndex, snmp$errorStatus, snmp$errorStatusText, snmp$nonRepeaters, snmp$requestID, snmp$type, snmp$variableBindings"),
-    @WritesAttribute(attribute=SNMPUtils.SNMP_PROP_PREFIX + "textualOid", description="This attribute will exist if and only if the strategy"
-            + " is GET and will be equal to the value given in Textual Oid property.")
+        @WritesAttribute(attribute = SNMPUtils.SNMP_PROP_PREFIX + "*", description = "Attributes retrieved from the SNMP response. It may include:"
+                + " snmp$errorIndex, snmp$errorStatus, snmp$errorStatusText, snmp$nonRepeaters, snmp$requestID, snmp$type, snmp$variableBindings"),
+        @WritesAttribute(attribute = SNMPUtils.SNMP_PROP_PREFIX + "textualOid", description = "This attribute will exist if and only if the strategy"
+                + " is GET and will be equal to the value given in Textual Oid property.")
 })
-public class GetSNMP extends AbstractSNMPProcessor<SNMPGetter> {
+public class GetSNMP extends AbstractSNMPProcessor {
+
+    // SNMP strategies
+    public static final AllowableValue GET = new AllowableValue("GET", "GET",
+            "A manager-to-agent request to retrieve the value of a variable. A response with the current value returned.");
+    public static final AllowableValue WALK = new AllowableValue("WALK", "WALK",
+            "A manager-to-agent request to retrieve the value of multiple variables. Snmpwalk also traverses all subnodes" +
+                    "under the specified OID.");
 
-    /** OID to request (if walk, it is the root ID of the request) */
+    // OID to request (if walk, it is the root ID of the request).
     public static final PropertyDescriptor OID = new PropertyDescriptor.Builder()
             .name("snmp-oid")
             .displayName("OID")
-            .description("The OID to request")
+            .description("Each OID (object identifier) identifies a variable that can be read or set via SNMP.")
             .required(true)
-            .addValidator(SNMPUtils.SNMP_OID_VALIDATOR)
+            .addValidator(new OIDValidator())
+            .build();
+
+    public static final PropertyDescriptor SNMP_STRATEGY = new PropertyDescriptor.Builder()
+            .name("snmp-strategy")
+            .displayName("SNMP Strategy (GET / WALK)")
```

Comment:
The options in the list show the possible values (GET and WALK). I think it can be removed from the displayName.